web/web.go

```diff
@@ -0,0 +1,1022 @@
+		for _, column := range fk.Columns {
+			if column == col {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// TODO: create a map to store referenced column to get
+// this information in O(1).
+func isReferencedByFK(col, table string) bool {
+	for _, spSchema := range app.conv.SpSchema {
+		if table != spSchema.Name {
+			for _, fk := range spSchema.Fks {
+				if fk.ReferTable == table {
+					for _, column := range fk.ReferColumns {
+						if column == col {
+							return true
+						}
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+func canRemoveColumn(colName, table string) (error, int) {
+	if isPartOfPK := isPartOfPK(colName, table); isPartOfPK {
+		if err := rollback(); err != nil {
+			return fmt.Errorf("rollback failed"), http.StatusInternalServerError
+		}
+		return fmt.Errorf("Column is part of primary key"), http.StatusBadRequest
+	}
+	if isPartOfSecondaryIndex := isPartOfSecondaryIndex(colName, table); isPartOfSecondaryIndex {
+		if err := rollback(); err != nil {
+			return fmt.Errorf("rollback failed"), http.StatusInternalServerError
+		}
+		return fmt.Errorf("Column is part of secondary index, remove secondary index before making the update"), http.StatusPreconditionFailed
+	}
+	isPartOfFK := isPartOfFK(colName, table)
+	isReferencedByFK := isReferencedByFK(colName, table)
+	if isPartOfFK || isReferencedByFK {
+		if err := rollback(); err != nil {
+			return fmt.Errorf("rollback failed"), http.StatusInternalServerError
+		}
+		return fmt.Errorf("Column is part of foreign key relation, remove foreign key constraint before making the update"), http.StatusPreconditionFailed
+	}
+	return nil, http.StatusOK
+}
+
+func canRenameOrChangeType(colName, table string) (error, int) {
+	isPartOfPK := isPartOfPK(colName, table)
+	isParent := isParent(table)
+	isChild := app.conv.SpSchema[table].Parent != ""
+	if isPartOfPK && (isParent || isChild) {
+		if err := rollback(); err != nil {
+			return fmt.Errorf("rollback failed"), http.StatusInternalServerError
+		}
+		return fmt.Errorf("Column is part of parent-child relation"), http.StatusBadRequest
+	}
+	if isPartOfSecondaryIndex := isPartOfSecondaryIndex(colName, table); isPartOfSecondaryIndex {
+		if err := rollback(); err != nil {
+			return fmt.Errorf("rollback failed"), http.StatusInternalServerError
+		}
+		return fmt.Errorf("Column is part of secondary index, remove secondary index before making the update"), http.StatusPreconditionFailed
+	}
+	isPartOfFK := isPartOfFK(colName, table)
+	isReferencedByFK := isReferencedByFK(colName, table)
+	if isPartOfFK || isReferencedByFK {
+		if err := rollback(); err != nil {
+			return fmt.Errorf("rollback failed"), http.StatusInternalServerError
+		}
+		return fmt.Errorf("Column is part of foreign key relation, remove foreign key constraint before making the update"), http.StatusPreconditionFailed
+	}
+	return nil, http.StatusOK
+}
+
+// TODO: improve canChangePK to include case for foreign key
```

Comment:
Let's either delete this function, or add a comment:

// This function is currently unused.

Reply:
done